cmd/kube-apiserver/app/server.go

```diff
@@ -551,37 +552,39 @@ func (s *APIServer) Run(_ []string) error {
 	}
 
 	config := &master.Config{
-		StorageDestinations:       storageDestinations,
-		StorageVersions:           storageVersions,
-		EventTTL:                  s.EventTTL,
-		KubeletClient:             kubeletClient,
-		ServiceClusterIPRange:     &n,
-		EnableCoreControllers:     true,
-		EnableLogsSupport:         s.EnableLogsSupport,
-		EnableUISupport:           true,
-		EnableSwaggerSupport:      true,
-		EnableProfiling:           s.EnableProfiling,
-		EnableWatchCache:          s.EnableWatchCache,
-		EnableIndex:               true,
-		APIPrefix:                 s.APIPrefix,
-		APIGroupPrefix:            s.APIGroupPrefix,
-		CorsAllowedOriginList:     s.CorsAllowedOriginList,
-		ReadWritePort:             s.SecurePort,
-		PublicAddress:             s.AdvertiseAddress,
-		Authenticator:             authenticator,
-		SupportsBasicAuth:         len(s.BasicAuthFile) > 0,
-		Authorizer:                authorizer,
-		AdmissionControl:          admissionController,
-		APIGroupVersionOverrides:  apiGroupVersionOverrides,
-		MasterServiceNamespace:    s.MasterServiceNamespace,
-		ClusterName:               s.ClusterName,
-		ExternalHost:              s.ExternalHost,
-		MinRequestTimeout:         s.MinRequestTimeout,
-		ProxyDialer:               proxyDialerFn,
-		ProxyTLSClientConfig:      proxyTLSClientConfig,
-		Tunneler:                  tunneler,
-		ServiceNodePortRange:      s.ServiceNodePortRange,
-		KubernetesServiceNodePort: s.KubernetesServiceNodePort,
+		Config: &genericapiserver.Config{
+			StorageDestinations:       storageDestinations,
+			StorageVersions:           storageVersions,
+			EventTTL:                  s.EventTTL,
+			KubeletClient:             kubeletClient,
+			ServiceClusterIPRange:     &n,
+			EnableLogsSupport:         s.EnableLogsSupport,
+			EnableUISupport:           true,
+			EnableSwaggerSupport:      true,
+			EnableProfiling:           s.EnableProfiling,
+			EnableWatchCache:          s.EnableWatchCache,
+			EnableIndex:               true,
+			APIPrefix:                 s.APIPrefix,
+			APIGroupPrefix:            s.APIGroupPrefix,
+			CorsAllowedOriginList:     s.CorsAllowedOriginList,
+			ReadWritePort:             s.SecurePort,
+			PublicAddress:             s.AdvertiseAddress,
+			Authenticator:             authenticator,
+			SupportsBasicAuth:         len(s.BasicAuthFile) > 0,
+			Authorizer:                authorizer,
+			AdmissionControl:          admissionController,
+			APIGroupVersionOverrides:  apiGroupVersionOverrides,
+			MasterServiceNamespace:    s.MasterServiceNamespace,
+			ClusterName:               s.ClusterName,
+			ExternalHost:              s.ExternalHost,
+			MinRequestTimeout:         s.MinRequestTimeout,
+			ProxyDialer:               proxyDialerFn,
+			ProxyTLSClientConfig:      proxyTLSClientConfig,
+			ServiceNodePortRange:      s.ServiceNodePortRange,
+			KubernetesServiceNodePort: s.KubernetesServiceNodePort,
```

Comment:
Maybe it's better to list the things I think *are* generic:
* StorageDestinations
* StorageVersions
* Swagger
* Profiling (maybe?)
* WatchCache
* Cors
* Read/write port
* public address
* auth things
* admission - maybe?
* group version overrides
* Master service namespace -- each set of apiservers will need to make their own service
* cluster name -- do all apiservers need to know this?
* MinRequestTimeout -- maybe
* proxy stuff -- probably

Things not mentioned are not generic IMO